### ISSUE TITLE
Add an API to pessimistically lock and unlock the iceberg table in catalog. 

### ISSUE THREAD

**Issue by vamen** [apache/iceberg#5607]
### Feature Request / Improvement

Though it is rare, but it often happens that spark gets shut down when the iceberg table write is still in progress and dangling lock is left in hive_locks table and subsequent run gets failed because of lock present in hive_lockcs table. I bet this issue happen occasionally in other tables which are used for locking.
In such cases we manually unlock the table using hive metastore client.

I think it is usefull if we have an api in catalog similar to create table, drop table etc. to unlock the table. 

Also we have a use case where we read the data from iceberg table modify it and write it back. in this process we don't want other process to write to the same iceberg table. it would be great if lock api is also supported at catalog level so that we can  pessimistically  lock the iceberg table.

So in summary it would be great if we can add support lock and unlock table apis at catalog  level.

This will solve common question that we often come across where people requires transaction support across commit in spark.

If there is initial consent  for this feature. then would like to put up a design and raise pr for same.
